eureka/S5_lightcurve_fitting/models/BatmanModels.py

```diff
@@ -0,0 +1,210 @@
+            
+            #Set limb darkening parameters
+            uarray=[]
+            for u in self.coeffs:
+                index=np.where(np.array(self.paramtitles)==u)[0]
+                if len(index)!=0:
+                    item=self.longparamlist[c][index[0]]
+                    uarray.append(self.parameters.dict[item][0])
+            bm_params.u = uarray
+
+            # Use batman ld_profile name
+            if self.parameters.limb_dark.value == '4-parameter':
+                bm_params.limb_dark = 'nonlinear'
+
+            # Make the transit model
+            m_transit = batman.TransitModel(bm_params, self.time, transittype='primary')
+
+            lcfinal = np.append(lcfinal, m_transit.light_curve(bm_params))
+
+        return lcfinal
+
+class EclipseModel(Model):
+    """Eclipse Model"""
+    def __init__(self, **kwargs):
+        """Initialize the eclipse model
+        """
+        # Inherit from Model calss
+        super().__init__(**kwargs)
+
+        # Define model type (physical, systematic, other)
+        self.modeltype = 'physical'
+
+        # Check for Parameters instance
+        self.parameters = kwargs.get('parameters')
+
+        # Generate parameters from kwargs if necessary
+        if self.parameters is None:
+            self.parameters = Parameters(**kwargs)
+        
+        # Set parameters for multi-channel fits
+        self.longparamlist = kwargs.get('longparamlist')
+        self.nchan = kwargs.get('nchan')
+        self.paramtitles = kwargs.get('paramtitles')
+
+        # Store the ld_profile
+        self.ld_func = ld_profile('uniform')
+
+    def eval(self, **kwargs):
+        """Evaluate the function with the given values"""
+        # Get the time
+        if self.time is None:
+            self.time = kwargs.get('time')
+        
+       #Initialize model
+        bm_params = batman.TransitParams()
+
+        # Set all parameters
+        lcfinal=np.array([])
+        for c in np.arange(self.nchan):
+            # Set all parameters
+            for index,item in enumerate(self.longparamlist[c]):
+                setattr(bm_params, self.paramtitles[index], self.parameters.dict[item][0])
+
+            bm_params.limb_dark = 'uniform'
+            bm_params.u = []
+
+            if not np.any(['t_secondary' in key for key in self.longparamlist[c]]):
+                # If not explicitly fitting for the time of eclipse, get the time of eclipse from the time of transit, period, eccentricity, and argument of periastron
+                m_transit = batman.TransitModel(bm_params, self.time, transittype='primary')
+                bm_params.t_secondary = m_transit.get_t_secondary(bm_params)
+            
+            # Make the eclipse model
+            m_eclipse = batman.TransitModel(bm_params, self.time, transittype='secondary')
+
+            lcfinal = np.append(lcfinal, m_eclipse.light_curve(bm_params))
+
+        return lcfinal
+
+class PhaseVariationsModel(Model):
```

Comment:
`PhaseCurveModel` makes more sense to me

Reply:
My hesitation with doing so (and for using phase above instead of phasecurve) was that to me phase curve implies the entirety of the phase curve (transit and eclipse included) while phase variations implies just the sinusoidal modulations to me. I can still change this and/or the above if you prefer though